src/SystemManager.hh

```diff
@@ -92,12 +92,26 @@ namespace gz
       /// \return The number of newly-active systems
       public: size_t ActivatePendingSystems();
 
-      /// \brief Get an vector of all active systems implementing "Configure"
-      /// \return Vector of systems's configure interfaces.
+      /// \brief Perform a reset on all systems
+      ///
+      /// If a system implements the ISystemReset interface, it will be called.
+      //
+      /// Otherwise, if a system does not have the ISystemReset interface
+      /// implemented, and was created via loading a plugin,
+      /// that plugin will be reloaded.
+      ///
+      /// Otherwise, if a system is created from in-memory rather than a plugin,
+      /// that system will remain unaffected.
```

Comment:
It is a limitation of the architecture of the in-memory path.  Once a plugin is created in-memory, we don't have any way to recreate that plugin.

With gz-plugin, we always have the capability to re-spawn plugins because we have access to the factory methods in the plugin library.

I think the correct answer would be to make the in-memory systems align more closely with shared-library systems by doing a registration system.  

In the near term, I can update the documentation to make this limitation more clear.